craft_parts/sequencer.py

```diff
@@ -190,30 +193,26 @@ def _run_step(
             state = states.PullState(
                 part_properties=part_properties,
                 project_options=self._project_info.project_options,
-                assets={},  # TODO: obtain pull assets
             )
 
         elif step == Step.BUILD:
             state = states.BuildState(
                 part_properties=part_properties,
                 project_options=self._project_info.project_options,
-                assets={},  # TODO: obtain build assets
+                overlay_hash=self._layer_state.get_overlay_hash().hex(),
             )
 
         elif step == Step.STAGE:
             state = states.StageState(
                 part_properties=part_properties,
                 project_options=self._project_info.project_options,
-                files=set(),
-                directories=set(),
+                overlay_hash=self._layer_state.get_overlay_hash().hex(),
```

Comment:
`overlay_hash` is not specified in the States; should it be?

Reply:
`overlay_hash` is specific to the Build and Stage states, which are the steps affected by changes in the overlay verification hash.

Reply:
But it's not specified in `BuildState` either, in `craft_parts/state_manager/build_state.py`....

Reply:
It is there:
```
class BuildState(StepState):
    """Context information for the build step."""

    assets: Dict[str, Any] = {}
    overlay_hash: Optional[str] = None
```
This was added in 96274b8

Reply:
Sorry, I didn't have the local copy updated :/